### DISCUSSION TITLE
TriggerConnections() does not terminate?

### DISCUSSION THREAD

**Discussion by troglobit** [EIPStackGroup/OpENer#387] [Q&A] [answered]
Hi,

am I missing something in this function, or is it missing a `node = node->next;` at the end of the `while()` loop?

If the first open connection doesn't match the output or input assembly arguments to the function we seem to be spinning forever on the first node. Right?

https://github.com/EIPStackGroup/OpENer/blob/329b231ec7c9d891046cb7858443a78dc0915889/source/src/cip/cipconnectionmanager.c#L1694-L1716

**Comment by MartinMelikMerkumians** [comment, marked as answer]
Hi,

shame on me. Yes seems an oversight by me. Guess that code has never been used in any tests 😅

**Reply by troglobit** [reply]
OK, I can send a PR later today to fix it.
